exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,305 @@
+  // finished.
+  auto connect_to_broker() -> asio::awaitable<void> {
+    while (true) {
+      try {
+        logger_.trace("Connecting to MQTT broker");
+        asio::ip::tcp::socket resolve_sock{ ctx_ };
+        asio::ip::tcp::resolver res{ resolve_sock.get_executor() };
+
+        logger_.trace("Resolving IP number");
+        asio::ip::tcp::resolver::results_type resolved_ip =
+            co_await res.async_resolve(mqtt_host_, mqtt_port_, asio::use_awaitable);
+
+        logger_.trace("Connecting MQTT client socket to IP number of broker");
+        co_await asio::async_connect(mqtt_client_->next_layer(), resolved_ip, asio::use_awaitable);
+
+        logger_.trace("Sending Connect request to the MQTT broker");
+
+        // Connect is initialized with clean start set to false, which means that the broker will keep the session alive even
+        // if the client disconnect, and it will keep that session alive for UINT_MAX seconds.
+        auto connect_packet =
+            async_mqtt::v5::connect_packet{ false,
+                                            std::chrono::seconds(100).count(),
+                                            async_mqtt::allocate_buffer("cid1"),
+                                            async_mqtt::nullopt,
+                                            async_mqtt::allocate_buffer(mqtt_username_),
+                                            async_mqtt::allocate_buffer(mqtt_password_),
+                                            { async_mqtt::property::session_expiry_interval{ UINT_MAX } } };
+
+        co_await mqtt_client_->send(connect_packet, asio::use_awaitable);
+
+        logger_.trace("Waiting for MQTT connection acknowledgement");
+        co_await mqtt_client_->recv(async_mqtt::filter::match, { async_mqtt::control_packet_type::connack },
+                                    asio::use_awaitable);
+
+        connect_active_ = false;
+        co_return;
+      } catch (std::exception& e) {
+        logger_.error("Error while connecting to MQTT broker: {}", e.what());
+      } catch (...) {
+        logger_.error("Unknown error while connecting to MQTT broker");
+      }
+
+      // If the connection failed, then clear the socket and try again after 1 second.
+      logger_.trace("Waiting for 1 second and then trying to connect again");
+      co_await asio::co_spawn(mqtt_client_->strand(), mqtt_client_->close(asio::use_awaitable), asio::use_awaitable);
+      co_await asio::steady_timer{ ctx_, std::chrono::seconds{ 1 } }.async_wait(asio::use_awaitable);
+    }
+  }
+
+  // Function which has oversight over the signals
+  auto load_signals() -> void {
+    // Stop reading the current signals by canceling the slots
+    for (auto& slot : slots_) {
+      std::visit([](auto& ptr) { ptr->cancel(); }, slot);
+    }
+
+    active_signals_.clear();
+    get_signals();
+    clean_signals();
+
+    for (tfc::ipc_ruler::signal& signal : active_signals_) {
+      handle_signal(signal);
```

Comment:
This is okay and could be misunderstood, when reading the code it is nice to know what is going on and `handle_signal` does not describe what will occur. 
I would propose make something like:
```c++
auto slot = make_connection(signal);
slots_.emplace_back(slot);
asio::co_spawn(slot...);
```

Reply:
Do you propose this change inside of run_slot, handle_signal or load_signals? the handle_signal function is at least needed to determine the type.

Reply:
this code has changed significantly so marking this as resolved.